src/Module.Server/Balancing/MatchBalancer.cs

```diff
@@ -415,6 +416,23 @@ private bool FindAndSwapUsers(GameMatch gameMatch, float sizeScaler)
         return (bestClanGroupToSwapSource, bestClanGroupToSwapDestination, distanceToTargetVector);
     }
 
+    private bool IsClanGroupsSwapValid(List<ClanGroup> strongTeam, List<ClanGroup> weakTeam, bool swappingFromWeakTeam,
+    int sourceGroupSize, float sourceGroupWeight, int destinationGroupSize, float destinationGroupWeight,
+    float sizeScaler)
+    {
+        float newTeamWeightDiff = swappingFromWeakTeam
+            ? strongTeam.Sum(c => c.Weight()) + 2f * sourceGroupWeight - 2f * destinationGroupWeight - weakTeam.Sum(c => c.Weight())
+            : strongTeam.Sum(c => c.Weight()) - 2f * sourceGroupWeight + 2f * destinationGroupWeight - weakTeam.Sum(c => c.Weight());
+        float newTeamSizeDiff = swappingFromWeakTeam
+            ? strongTeam.Count + 2 * sourceGroupSize - 2f * destinationGroupSize - weakTeam.Count
+            : strongTeam.Count - 2 * sourceGroupSize + 2f * destinationGroupSize - weakTeam.Count;
+
+        Vector2 oldDifferenceVector = new((strongTeam.Count - weakTeam.Count) * sizeScaler,
+            strongTeam.Sum(c => c.Weight()) - weakTeam.Sum(c => c.Weight()));
+        Vector2 newDifferenceVector = new(newTeamSizeDiff * sizeScaler, newTeamWeightDiff);
+        return newDifferenceVector.Length() < oldDifferenceVector.Length();
+    }
+
     private bool IsSwapValid(List<WeightedCrpgUser> strongTeam, List<WeightedCrpgUser> weakTeam, bool swappingFromWeakTeam,
```

Comment:
Is this method still used?

Reply:
needed by a method that is bugged and need a fix ( FindAndSwapUsers )